File: C++/Approach_3_O(n)_Time_&_O(1)_Space.cpp

// O(n) time | O(1) space

using namespace std;

int getNthFib(int n) {
  int lastTwo[] = {0, 1};
  int counter = 3;
  while (counter <= n) {
    int nextFib = lastTwo[0] + lastTwo[1];
    lastTwo[0] = lastTwo[1];
    lastTwo[1] = nextFib;
    counter++;
  }
  return n > 1 ? lastTwo[1] : lastTwo[0];
}

File: Swift/Approach_3_O(n)_Time_&_O(1)_Space.swift

​// O(n) time | O(1) space

func getNthFib(n: Int) -> Int {
    var lastTwo = [0, 1]
    var counter = 3
    while counter <= n {
        let nextFib = lastTwo[0] + lastTwo[1]
        lastTwo[0] = lastTwo[1]
        lastTwo[1] = nextFib
        counter = counter + 1
    }
    return n > 1 ? lastTwo[1] : lastTwo[0]
}
